### DISCUSSION TITLE
How to prevent KEDA from leaving only 1 Pod as Ready when performing a RollingUpdate?

### DISCUSSION THREAD

**Discussion by jpereiramp** [kedacore/keda#2713] [Q&A / Need Help]
Hi all, happy Friday!

I have a question regarding Rolling Updates on `Deployments` when using a KEDA `ScaledObject` attached to such deployment, with `minReplicas: 3`. 

I have configured my `Deployment` with a `RollingUpdate` strategy defining `maxUnavailable: 0`. However, whenever I update such deployment, for example set a new environment variable, I expect that this configuration should be respected, and only one `Pod` at a time is updated. I also expect that the current desired replicas should be respected, and not scaled-down to for example the `minReplicas` configuration.

What I am currently seeing is that only 1 Pod is left available and then KEDA scales-up to at least the minReplica count because it recognizes that the current available replicas is below the set `minReplicas`.

If I configure my auto-scaling with a plain `HorizontalPodAutoscaler` object (using Prometheus Adapter as a bridge for my custom metrics), I get the behavior that I expect. But switching to KEDA, I lose that and am left with only 1 Running and Ready `Pod` whenever I do a `RollingUpdate`.

Does anyone know how I can achieve that behavior? Thanks in advance!

**TLDR: When using KEDA attached to a Deployment, my RollingUpdate strategy is not respected, more specifically my maxUnavailable setting.**

**Comment by yuvalweber** [comment]
hey @jpereiramp can you please provide your scaledObject + deployment?
